core/src/main/java/org/apache/iceberg/TableMetadata.java

```diff
@@ -1328,5 +1387,16 @@ private void setBranchSnapshot(Snapshot snapshot, String branch, Long currentTim
 
       return newSnapshotLog;
     }
+
+    private boolean isAddedSnapshot(long snapshotId) {
+      return changes(MetadataUpdate.AddSnapshot.class)
+          .anyMatch(add -> add.snapshot().snapshotId() == snapshotId);
+    }
+
+    private <U extends MetadataUpdate> Stream<U> changes(Class<U> updateClass) {
+      return changes.stream()
+          .filter(updateClass::isInstance)
+          .map(updateClass::cast);
```

Comment:
If this is the only place this is used, is it really needed? If so, maybe `changesFor(Class<U> updateClass)` would be better. Specifically as ther eis already a class field called `changes`.

Reply:
There are a few places where we go through the changes, so I think it's a good idea to keep this for future changes rather than implementing it slightly differently in multiple places.